cardano-api/src/Cardano/Api/Tx.hs

```diff
@@ -933,6 +945,7 @@ signByronTransaction nw txbody sks =
 
 -- signing keys is a set
 signShelleyTransaction :: IsShelleyBasedEra era
+                       => Ledger.HashAnnotated (Ledger.TxBody (ShelleyLedgerEra era)) Ledger.EraIndependentTxBody StandardCrypto
```

Comment:
Likewise, we don't want to propagate constraints at the top level

Reply:
Done!